velox/functions/prestosql/HyperLogLogFunctions.h

```diff
@@ -81,4 +84,178 @@ struct EmptyApproxSetWithMaxErrorFunction {
     return true;
   }
 };
+
+class MergeHllVectorFunction : public exec::VectorFunction {
+ public:
+  bool isDefaultNullBehavior() const {
+    return true;
+  }
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    // Create a memory pool for the entire batch
+    auto pool = memory::memoryManager()->addLeafPool();
```

Comment:
@natashasehgal We do not want to create separate memory pools for each batch of data. Let's discuss over VC how to proceed.